### PR TITLE
fix: add a way to run page.$$ without isolation

### PR DESCRIPTION
**Pull request by OrKoN**
With this change, almost all of the performance regression is recovered. It requires the client to specify `isolate: false` when using `page.$$` to opt out of query isolation.

Related to https://github.com/puppeteer/puppeteer/issues/11944 #12566

### CODE REVIEW

**Comment by OrKoN** [comment]
```
Node: v20.10.0
Puppeteer: 21.1.1
Browser: HeadlessChrome/116.0.5845.96
Duration: 40.403416991233826 (without isolation)
```

```
Node: v20.10.0
Puppeteer: 21.2.0
Browser: HeadlessChrome/116.0.5845.96
Duration: 128.20825004577637 (with isolation)
```

```
Node: v20.10.0
Puppeteer: 22.10.0
Browser: HeadlessChrome/125.0.6422.141
Duration: 187.64925003051758 (with isolation)
```

```
Node: v20.10.0
Puppeteer: 22.10.0
Browser: HeadlessChrome/125.0.6422.141
Duration: 79.55983293056488 (without isolation)
```

File: test/src/queryselector.spec.ts [hunk 1]
```diff
@@ -153,6 +153,24 @@ describe('querySelector', function () {
     });
   });
 
+  describe('Page.$$s', function () {
```
Comment (Lightning00Blade) [inline test/src/queryselector.spec.ts]: Let's move it under the `$$` tests and update the `it` text

File: packages/puppeteer-core/src/api/ElementHandle.ts [hunk 2]
```diff
@@ -371,15 +375,43 @@ export abstract class ElementHandle<
...
-      QueryHandler.queryAll(this, updatedSelector)
-    ) as Promise<Array<ElementHandle<NodeFor<Selector>>>>);
+    if (options?.isolate === false) {
+      const {updatedSelector, QueryHandler} =
+        getQueryHandlerAndSelector(selector);
+      return await (AsyncIterableUtil.collect(
+        QueryHandler.queryAll(this, updatedSelector)
+      ) as Promise<Array<ElementHandle<NodeFor<Selector>>>>);
+    }
+    // If the handle is already isolated, then we don't need to adopt it
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/ElementHandle.ts]: WDYT about have a private method and attaching the decorator to it then calling it here?

Comment (OrKoN) [reply packages/puppeteer-core/src/api/ElementHandle.ts]: didn't get the idea with the private method but let's reuse the decorator

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**